app/src/main/java/ch/epfl/sdp/contamination/fragment/UserInfectionFragment.java

```diff
@@ -177,58 +199,46 @@ private void setOnlineOfflineVisibility(boolean isOnline) {
         infectionStatusView.setVisibility(onlineVisibility);
     }
 
-    private void getLoggedInUser() {
-        account = AuthenticationManager.getAccount(getActivity());
-        userName = account.getDisplayName();
-        retrieveUserInfectionStatus().thenAccept(this::setInfectionColorAndMessage);
-    }
 
     private boolean checkElapsedTimeSinceLastChange() {
-        Date currentTime = Calendar.getInstance().getTime();
-        //TODO: what does this means?
-        /* get 1 jan 1970 by default. It's definitely wrong but works as we want t check that
-         * the status has not been updated less than a day ago.
-         */
-        Date lastStatusChange = new Date(sharedPref.getLong("lastStatusChange", 0));
-        long difference = Math.abs(currentTime.getTime() - lastStatusChange.getTime());
-        long differenceDays = difference / (24 * 60 * 60 * 1000);
+        if (IS_DEMO) {
+            return true;
+        } else {
+            Date currentTime = Calendar.getInstance().getTime();
+            //TODO: what does this means?
+            //get 1 jan 1970 by default. It's definitely wrong but works as we want t check that
+            //the status has not been updated less than a day ago.
 
-        sharedPref.edit().putLong("lastStatusChange", currentTime.getTime()).apply();
-        return differenceDays > 1;
+            Date lastStatusChange = new Date(sharedPref.getLong("lastStatusChange", 0));
+            long difference = Math.abs(currentTime.getTime() - lastStatusChange.getTime());
+            long differenceDays = difference / (24 * 60 * 60 * 1000);
+
+            sharedPref.edit().putLong("lastStatusChange", currentTime.getTime()).apply();
+            return differenceDays > 1;
+        }
     }
 
     private void executeHealthStatusChange() {
         CharSequence buttonText = infectionStatusButton.getText();
         boolean infected = buttonText.equals(getResources().getString(R.string.i_am_infected));
         if (infected) {
-            //Tell the analyst we are now sick !
-            service.getAnalyst().updateStatus(Carrier.InfectionStatus.INFECTED);
+            service.getAnalyst().getCarrier().evolveInfection(new Date(), INFECTED, 1f);
             setInfectionColorAndMessage(true);
-            modifyUserInfectionStatus(userName, true);
         } else {
-            //Tell analyst we are now healthy !
-            service.getAnalyst().updateStatus(Carrier.InfectionStatus.HEALTHY);
+            service.getAnalyst().getCarrier().evolveInfection(new Date(), HEALTHY, 0f);
             sendRecoveryToFirebase();
             setInfectionColorAndMessage(false);
-            modifyUserInfectionStatus(userName, false);
         }
     }
 
     private void sendRecoveryToFirebase() {
+        // TODO: @Lucie I think that doesn't upload anything to Firestore
```

Comment:
Actually for me it worked (it's me who wrote this). There is a test that checks exactly that.